examples/zephyr/http-client/Makefile

```diff
@@ -1,59 +1,40 @@
-MONGOOSE_ROOT ?= $(realpath $(CURDIR)/../../..)
-DOCKER_PROJECT_DIR ?= /workdir
-ZEPHYR_DIR ?= zephyrproject
-PROJECT_NAME = $(notdir $(CURDIR))
-PROJECT_PATH = $(realpath $(CURDIR))
-ZEPHYR_PATH = $(realpath $(CURDIR)/..)/$(ZEPHYR_DIR)
-DOCKER_PROJECT_PATH = $(DOCKER_PROJECT_DIR)/$(PROJECT_NAME)
-DOCKER_ZEPHYR_PATH = $(DOCKER_PROJECT_DIR)/$(ZEPHYR_DIR)
-
-BOARD ?= qemu_x86
-ifeq "$(BOARD)" "qemu_x86"
-OVERLAY ?= -DOVERLAY_CONFIG=overlay-e1000.conf
-else
-OVERLAY ?= 
-endif
-
-SHAREUSB ?= --privileged -v /dev/bus/usb/:/dev/bus/usb
-SHARETUN ?= --cap-add=NET_ADMIN --device /dev/net/tun --net=host
-DOCKER ?= docker run --rm -v $(PROJECT_PATH):$(DOCKER_PROJECT_PATH) -v $(ZEPHYR_PATH):$(DOCKER_ZEPHYR_PATH)
-REPO ?= zephyrprojectrtos/ci
-
-
-example:
+TOP_DIR ?= $(realpath $(CURDIR)/../../..)
+ZEPHYR_DIR ?= $(TOP_DIR)/zephyrproject
+DOCKER ?= docker run --rm \
+          -v $(TOP_DIR):$(TOP_DIR) \
+          -w $(realpath $(CURDIR)) \
+          -e ZEPHYR_BASE=$(ZEPHYR_DIR)/zephyr \
+          $(DOCKER_ARGS) zephyrprojectrtos/ci
+#BUILD_ARGS ?= -b qemu_x86 -DOVERLAY_CONFIG=overlay-e1000.conf
+BUILD_ARGS ?= -b nucleo_f429zi
+
+.PHONY: build flash clean pristine run
+
+example zephyr:
 	true
 
-build:
-	cp $(MONGOOSE_ROOT)/mongoose.[c,h] src/
-	$(DOCKER) $(REPO) /bin/sh -c 'cd $(DOCKER_ZEPHYR_PATH)/zephyr && \
-	west build -b $(BOARD) -p auto $(DOCKER_PROJECT_PATH) $(OVERLAY) --build-dir $(DOCKER_PROJECT_PATH)/build'
+$(ZEPHYR_DIR):
+	mkdir -p $@
+	$(DOCKER) west init $@
+	$(DOCKER) sh -c 'cd $@ && west update'
 
+build: $(ZEPHYR_DIR)
+	cp $(TOP_DIR)/mongoose.[c,h] src/
+	$(DOCKER) west build $(BUILD_ARGS) -p auto $(realpath $(CURDIR))
+
+run: DOCKER_ARGS ?= --cap-add=NET_ADMIN --device /dev/net/tun --net=host -it
 run:
-	$(DOCKER) -it $(SHARETUN) $(REPO) /bin/sh -c 'cd $(DOCKER_ZEPHYR_PATH) && \
-	west build -t run --build-dir $(DOCKER_PROJECT_PATH)/build'
+	$(DOCKER) west build -t run
```

Comment:
This won't do anything on an F429.
However, on a qemu board, if we can setup the tun/tap in the hosting Ubuntu at Github, qemu is built into the Zephyr Docker image, so we can run on an emulated board and actually test the example works without having to have an actual board connected to the Internet and flash it and etc.
Actually, this fake board was great for spotting the null ptr dereferences in http-client I fixed later, the 386 MMU raised an exception and QEMU showed a nice crash report that led me to the problem. You can also debug on this virtual board running a gdb server in QEMU and a gdb client on another instance of the Docker image (to use the same gdb version)

Reply:
I think using "BUILD_ARGS" we still can build / test with qemu target?

Reply:
Yes, I thought on integrating those tests on the Github pipeline, we can add that in the make command I guess